app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/TransportController.kt

```diff
@@ -0,0 +1,176 @@
+     * containing a NULL for the station and an empty string for the station id. This is not cached or saved to the database.
+     *
+     * @param widgetId The id of the widget
+     * @return The WidgetDepartures Object
+     */
+    fun getWidget(widgetId: Int): WidgetDepartures {
+        if (widgetDeparturesList.indexOfKey(widgetId) >= 0) {
+            return widgetDeparturesList.get(widgetId)
+        }
+        val widgetDepartures = WidgetDepartures().apply {
+            transportDao.getAllWithId(widgetId)?.let {
+                station = it.station
+                stationId = it.stationId
+                useLocation = it.location
+                autoReload = it.reload
+            }
+        }
+
+        widgetDeparturesList.put(widgetId, widgetDepartures)
+        return widgetDepartures
+    }
+
+    override fun getCards(cacheControl: CacheControl): List<Card> {
+        val results = ArrayList<Card>()
+        if (!NetUtils.isConnected(mContext)) {
+            return results
+        }
+
+        // Get station for current campus
+        val locMan = LocationManager(mContext)
+        val station = locMan.getStation() ?: return results
+
+        val departures = getDeparturesFromExternal(mContext, station.id)
+        val card = MVVCard(mContext).apply {
+            setStation(station)
+            setDepartures(departures)
+        }
+        results.add(card.getIfShowOnStart()!!)
+
+        return results
+    }
+
+    companion object {
+        private var widgetDeparturesList = SparseArray<WidgetDepartures>()
+
+        /**
+         * Get all departures for a station.
+         *
+         * @param stationID Station ID, station name might or might not work
+         * @return List of departures
+         */
+        fun getDeparturesFromExternal(context: Context, stationID: String): List<Departure> {
+            try {
+                val departures = MvvClient.getInstance(context)
+                        .getDepartures(stationID).execute().body() ?: return emptyList()
+                val departureList = departures.departureList ?: emptyList()
+
+                return departureList.map { (servingLine, dateTime, countdown) ->
+                    Departure(servingLine.name,
+                            servingLine.direction,
+                            servingLine.symbol,
+                            countdown,
+                            dateTime)
+                }.sortedBy { it.countDown }
+            } catch (e: IOException) {
+                Utils.log(e)
+            }
+
+            return emptyList()
+        }
+
+        /**
+         * Find stations by station name prefix
+         *
+         * @param prefix Name prefix
+         * @return List of StationResult
+         */
+        fun getStationsFromExternal(context: Context, prefix: String): List<StationResult> {
+            try {
+                val (results) = MvvClient.getInstance(context).getStations(prefix).execute().body()
```

Comment:
Why the parentheses?

Reply:
```kotlin
// This is a MvvStationList
val results = (...)

// results now is that MvvStationList unpacked into a List<StationResult>
val (results) = (...)
```

We actually receive a `MvvStationList` to trigger parsing by the `MvvStationListSerializer`, but the station list is just a wrapper around `List<StationResult>`, so we need to unpack that list